src/grt/grt-wave_options-parse.adb

```diff
@@ -0,0 +1,323 @@
+         Error_Context (Msg_Invalid_Format);
+      end if;
+
+      -- Catch version a.2
+      Num := Value (Line (First .. Dot_Index - 1));
+      if Num = -1 then
+         Error_Context (Msg_Invalid_Format);
+      end if;
+      Version.Major := Num;
+
+      -- Catch version 1.a
+      Num := Value (Line (Dot_Index + 1 .. Line'Last));
+      if Num = -1 then
+         Error_Context (Msg_Invalid_Format);
+      end if;
+      Version.Minor := Num;
+
+      if Version.Major /= Current_Version.Major
+        or else Version.Minor > Current_Version.Minor
+      then
+         Print_Context (Error);
+         Error_C ("unsupported format version; it must be ");
+         if Current_Version.Minor /= 0 then
+            Error_C ("between ");
+            Print_Version (Version_Type'(Current_Version.Major, 0));
+            Error_C (" and ");
+         end if;
+         Print_Version (Current_Version);
+         Error_E;
+      end if;
+
+   end Parse_Version;
+
+   procedure Print_Version (Version : Version_Type) is
+      Num_Str : String (1 .. Value_String_Size);
+      First : Positive;
+   begin
+      To_String (Num_Str, First, Ghdl_I32 (Version.Major));
+      Report_C (Num_Str (First .. Num_Str'Last));
+      Report_C (".");
+      To_String (Num_Str, First, Ghdl_I32 (Version.Minor));
+      Report_C (Num_Str (First .. Num_Str'Last));
+   end Print_Version;
+
+   --------------------------------------------------------------------------
+
+   procedure Parse_Path (Line : String_Acc) is
+      First, Last : Positive;
+      Tree_Updated : Boolean;
+      Tree_Index : Tree_Index_Type;
+   begin
+      To_Lower (Line_Context.Str.all);
+      Last := Line'First;
+      if Line (Line'First) = '/' then
+         Tree_Index := Entity;
+         Last := Last + 1;
+      else
+         Tree_Index := Pkg;
+      end if;
+      Tree_Cursor := Trees (Tree_Index);
+      Previous_Tree_Cursor := null;
+
+      loop
+         First := Last;
+
+         loop -- Find next identifier
+            if Line (Last) = Sep (Tree_Index) then
+               Last := Last - 1;
+               exit;
+            elsif Last = Line'Last then
+               exit;
+            end if;
+            Last := Last + 1;
+         end loop;
+
+         Check_Validity (Line (First .. Last));
+         Tree_Updated := Update_Tree (Line (First .. Last), Tree_Index);
+         Line_Context.Max_Level := Line_Context.Max_Level + 1;
+
+         if Last = Line'Last then
+            if not Tree_Updated then
+               Error_Context ("ignored already known signal path", Warning);
+            end if;
+            return;
+         end if;
+
+         Last := Last + 2; -- Skip the separator
+         if Last > Line'Last then -- catch signal paths ending with /
+            Error_Context ("invalid signal path");
+         end if;
+
+      end loop;
+
+   end Parse_Path;
+
+   function Update_Tree (Elem_Name : String; Tree_Index : Tree_Index_Type)
+            return Boolean is
```

Comment:
I have never look on the STYLE file before. Now I am correcting my code 
accordingly.


Le 23. 07. 16 à 12:18, tgingold a écrit :
>
> In src/grt/grt-wave_options-parse.adb 
> <https://github.com/tgingold/ghdl/pull/121#discussion_r71970615>:
>
> > +               Error_Context ("ignored already known signal path", Warning);
> > +            end if;
> > +            return;
> > +         end if;
> > +
> > +         Last := Last + 2; -- Skip the separator
> > +         if Last > Line'Last then -- catch signal paths ending with /
> > +            Error_Context ("invalid signal path");
> > +         end if;
> > +
> > +      end loop;
> > +
> > +   end Parse_Path;
> > +
> > +   function Update_Tree (Elem_Name : String; Tree_Index : Tree_Index_Type)
> > +            return Boolean is
>
> 'is' on a separate line.
> The style rules for subprograms are not simple, but they are 
> documented in the STYLE file.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/tgingold/ghdl/pull/121/files/4bf3293563ba8e95cc35992d029ff26c4f4e67e5#r71970615>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ATNePkfxKIsxROP_PymbCIvaMYEPvZE9ks5qYep7gaJpZM4JTTAB>.
>